mm/slab.c

```diff
@@ -3231,7 +3231,7 @@ slab_alloc_node(struct kmem_cache *cachep, gfp_t flags, int nodeid, size_t orig_
 
 	ptr = kfence_alloc(cachep, orig_size, flags);
 	if (ptr)
-		goto leave;
+		return ptr;
```

Comment:
I think we have a problem because we executed slab_pre_alloc_hook, but now won't execute the post_alloc_hook. I think the problem might have to do with memcgs, which do something both in the pre- and post-alloc hooks.

I think we require some symmetry here, but still am not entirely sure what the right solution is: either don't execute pre-/post- hooks or execute both; but I think we must execute the pre-hook, so that really leaves only 1 option.

Reply:
Wait, we're skipping slab_pre_alloc_hook() exactly here, so after landing this change the problem should be gone.
I can imagine however that if someone adds extra object setup to those hooks we'll have to execute them.

Reply:
slab_pre_alloc_hook is above at line 3228. We are skipping over slab_post_alloc_hook.

Reply:
Ah, sorry, you are right. Ok, let's try to preserve both pre_ and post_alloc_hook then.